exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -42,6 +43,10 @@ func (r *AzureManagedMachinePool) Default(client client.Client) {
 	if r.Spec.Name == nil || *r.Spec.Name == "" {
 		r.Spec.Name = &r.Name
 	}
+
+	if r.Spec.OsDiskType == nil || to.String(r.Spec.OsDiskType) == "" {
+		r.Spec.OsDiskType = to.StringPtr(string(containerservice.OSDiskTypeManaged))
```

Comment:
It looks like you added a default in the OpenAPI schema, can we remove this?

Reply:
Yes, thanks!

Because the schema is now injecting the default value, are we able to use validation to fail if nil or empty string?